docs-site/content/guide/docsearch.md

```diff
@@ -22,6 +22,12 @@ Follow the official [DocSearch documentation](https://docsearch.algolia.com/docs
 
 You can use one of those as templates to create your own `config.js`, pointing to your documentation site.
 
+#### Key Concepts
+
+- Your main text content, which is usually within a `<main>` or an `<article>` HTML element, should be identifiable with a class such as `.main`
```

Comment:
@ArsalaBangash I don't think this is a hard requirement. IIRC, docsearch lets you use any css selectors (not just class names) to select any part of the DOM to be used for lvl0-6 and text content. 

For eg: here's what I use for the Vuepress-based Typesense docs site: https://github.com/typesense/typesense-docsearch-scraper/blob/7a2f05831bc5569e6c19e311125cb0ce91c61b1f/configs/public/typesense_docs.json#L15-L20